### ISSUE TITLE
Unable to reproduce the GPT 2 small results

### ISSUE THREAD

**Issue by pmpalang** [Liuhong99/Sophia#43]
Hello forum,

I'm trying to reproduce the GPT2 small results using the commands/configurations in the repository. However, the trends are quite different in comparison to the one in the paper/README. For example, the validation curve is expected to provide 2X faster convergence, but I see that the they are almost on top of each other at 10k iterations. The only change from the original commands is that I have --nproc_per_node=4 and batch_size=2 to fit my GPU. I'm copying the validation loss and train loss vs iterations plots here for reference. Note, that the run is still in progress but even the trends are unlike what we expect from the paper/README. I would appreciate any help in enabling me to reproduce the original plots. 

Thanks,
Poovaiah

Tagging another similar issue #30 here.
![ValidationLoss](https://github.com/Liuhong99/Sophia/assets/98763718/e3f52da0-6aea-4d1f-bddc-0b5a41c171d8)
![Trainloss_without_smoothing](https://github.com/Liuhong99/Sophia/assets/98763718/b544a2b5-af89-4b29-8f44-04c723534f1c)
![Trainloss_with_smoothing](https://github.com/Liuhong99/Sophia/assets/98763718/8d3d2d38-7535-4d7b-9a05-43df19392bf2)

**Comment by Liuhong99** [comment]
If you were able to zoom in into the y-axis, you can observe that Sophia-G has 0.04 smaller validation loss than Adam. If you change the number of steps of the Adam run to 200k, you will find out Adam 200k is better than Adam 100K by 0.04.
